src/cow-react/modules/wallet/web3-react/connectors/Injected/index.tsx

```diff
@@ -0,0 +1,204 @@
+            // if we're here, we can try to add a new network
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            return this.provider!.request<void>({
+              method: 'wallet_addEthereumChain',
+              params: [{ ...desiredChainIdOrChainParameters, chainId: desiredChainIdHex }],
+            })
+          }
+
+          throw error
+        })
+    }
+
+    return Promise.all([
+      this.provider.request({ method: 'eth_chainId' }) as Promise<string>,
+      this.provider.request({ method: 'eth_requestAccounts' }) as Promise<string[]>,
+    ]).then(([chainId, accounts]) => {
+      const receivedChainId = this.parseChainId(chainId)
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      if (!desiredChainId || desiredChainId === receivedChainId) {
+        return this.actions.update({ chainId: receivedChainId, accounts })
+      }
+
+      const desiredChainIdHex = `0x${desiredChainId.toString(16)}`
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      return this.provider!.request<void>({
+        method: 'wallet_switchEthereumChain',
+        params: [{ chainId: desiredChainIdHex }],
+      }).catch(async (error: ProviderRpcError) => {
+        console.log('err!', error)
+        console.log('debug err', error)
+
+        if (error.code === 4902 && typeof desiredChainIdOrChainParameters !== 'number') {
+          // if we're here, we can try to add a new network
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          return this.provider!.request<void>({
+            method: 'wallet_addEthereumChain',
+            params: [{ ...desiredChainIdOrChainParameters, chainId: desiredChainIdHex }],
+          })
+        }
+
+        throw error
+      })
+    })
+  }
+
+  /** {@inheritdoc Connector.connectEagerly} */
+  public async connectEagerly(): Promise<void> {
+    this.isomorphicInitialize()
+
+    if (!this.provider) return
+
+    return Promise.all([
+      this.provider.request({ method: 'eth_chainId' }) as Promise<string>,
+      this.provider.request({ method: 'eth_accounts' }) as Promise<string[]>,
+    ])
+      .then(([chainId, accounts]) => {
+        if (accounts.length) {
+          this.actions.update({ chainId: this.parseChainId(chainId), accounts })
+        } else {
+          throw new Error('No accounts returned')
+        }
+      })
+      .catch((error) => {
+        console.debug('Could not connect eagerly', error)
+        this.actions.resetState()
+      })
+  }
+
+  private detectProvider(): InjectedWalletProvider | void {
+    const w = window as unknown as Window
+
+    this.provider = this.detectOnEthereum(w.ethereum) || this.detectOnProvider(w.ethereum?.providers) || null
+    ;(window as any)['provider'] = this.provider
+
+    return this.provider
+  }
+
+  private detectOnProvider(providers: any) {
```

Comment:
`any` => `import { ExternalProvider } from '@ethersproject/providers'`

Reply:
Doesn't really work in this case 😞

Reply:
https://github.com/cowprotocol/cowswap/pull/2257
This should work